corehq/apps/app_manager/suite_xml/sections/remote_requests.py

```diff
@@ -145,6 +145,13 @@ def _get_remote_request_query_datums(self):
             QueryData(key="{}".format(c.property), ref="{}".format(c.defaultValue))
             for c in self.module.search_config.default_properties
         ]
+        if self.module.search_config.search_filter:
+            extra_query_datums.append(
+                QueryData(
+                    key=CASE_SEARCH_XPATH_QUERY_KEY,
+                    ref="'{}'".format(self.module.search_config.search_filter)
```

Comment:
why is this surrounded in `' '`? Does that stop it from being evaluated on the phone? What happens if your xpath expression is something like `mood = 'heavy'`? If that's the case, what happens if the expression is `mood = 'heavy`?

Reply:
> Does that stop it from being evaluated on the phone?

I think this is the reason - without the quotes formplayer throws [this error](https://github.com/dimagi/commcare-core/blob/e555343ae81b330ef6e670754b4825e57bc4e3df/src/main/java/org/javarosa/xpath/expr/XPathPathExpr.java#L210).

> What happens if your xpath expression is something like mood = 'heavy'

Good point, I'll look into escaping.

Reply:
Hmm, I haven't thought through this all the way, but I don't think you want to escape everything here. I think the beauty of this is that you could use session data to enrich your search filter, e.g:
```
mood = instance('commcaresession')/session/user/data/current_mood
```
But I think that gets really complicated because it's confusing which parts should be evaluated server side and which parts are evaluated on the phone. It gets even more confusing with your "copy" button because not everything valid in a case filter expression will be valid on the server (like the above)

Maybe if the user is forced to write an expression that returns the xpath filter themselves:
```
concat('mood = ', instance('commcaresession')/session/user/data/current_mood)
```
But that is a bit silly.

Maybe in the general case you surround everything in `' '`, but you give people the option to write an "advanced" filter which interpolates things evaluated on the phone, and things evaluated on the server. In this case, you have to have a good mental model of what is happening behind the scenes, which would work for "advanced" users, but would leave the general case mostly accessible. 

Sorry to not have thought of this before...

Reply:
Hmm. Thanks for talking through this. Given that this is being written for the COVID projects specifically to use their case list filters as case claim filters, I'll take a look at how complicated their case list filters are.

Reply:
I'm thinking about setting this up as two filters: a "pre" filter that's a parameter to the case search and a "post" filter that's done on the phone. The projects that will use this feature do use filters that need to be evaluated on the phone (session and usercase).

The pre would use the code in this PR, while the post would be added to the remote request's datum's nodeset (the one that references the `results` instance), which has the session and I think the casedb available. The pre filter would make the search faster and return fewer results, and then the post filter would have the same abilities as case list filters.

I'm going to chat with the project team as well, but curious if you have thoughts on this. It might also be overly complicated and it could be better to just change this PR to filter post-search instead of pre-search.

Reply:
And I do agree the button to copy case list filter is confusing right now.

Reply:
I'm not sure I follow how the pre- and post filters work with what I was saying here. You mean that you would only be able to do "simple" filtering (without any evaluation of the xpaths) in a "pre" filter? 
It's a shame that I removed this: https://github.com/dimagi/commcare-hq/pull/28278 a few months ago, as I think it would have served your usecase. 
Maybe we can jump on a call and discuss this a bit? Even though you are building this for a specific usecase, I'd like to make sure we aren't closing doors on what could be a nice piece of functionality.

Reply:
Sure, let's chat about this, I'll set up time (might not be until next week).

Reply:
Updated in a7a1f1d8df0